src/coreclr/jit/gentree.cpp

```diff
@@ -20828,9 +20865,14 @@ GenTree* Compiler::gtNewSimdBinOpNode(
                 case TYP_ULONG:
                 {
                     assert((simdSize == 16) || (simdSize == 32) || (simdSize == 64));
-                    assert(compIsaSupportedDebugOnly(InstructionSet_AVX512DQ_VL));
+                    assert(compIsaSupportedDebugOnly(InstructionSet_AVX512DQ_VL) ||
+                           compIsaSupportedDebugOnly(InstructionSet_AVX10v1));
 
-                    if (simdSize != 64)
+                    if (simdSize != 64 && compOpportunisticallyDependsOn(InstructionSet_AVX10v1))
+                    {
+                        intrinsic = NI_AVX10v1_MultiplyLow;
+                    }
+                    else if (simdSize != 64)
```

Comment:
This can probably be something more like:
```cpp
if (simdSize == 64)
{
    assert(compIsaSupportedDebugOnly(InstructionSet_AVX512DQ));
    intrinsic = NI_AVX512DQ_MultiplyLow;
}
else if (compOpportunisticallyDependsOn(InstructionSet_AVX10v1))
{
    intrinsic = NI_AVX10v1_MultiplyLow;
}
else
{
    assert(compIsaSupportedDebugOnly(InstructionSet_AVX512DQ_VL));
    intrinsic = NI_AVX512DQ_VL_MultiplyLow;
}
```

It's a bigger diff from the baseline, but simplifies the overall checks/logic and intent.

Reply:
True. Made the changes here.